compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureDesugar.java

```diff
@@ -1609,8 +1609,7 @@ private static int findResolvedLevel(SymbolEnv symbolEnv, BVarSymbol varSymbol)
         SymbolEnv resolvedSymbolEnv = symbolEnv;
         while (resolvedSymbolEnv != null && resolvedSymbolEnv.node.getKind() != NodeKind.PACKAGE) {
             Scope.ScopeEntry entry = resolvedSymbolEnv.scope.lookup(varSymbol.name);
-            if (entry != NOT_FOUND_ENTRY && varSymbol == entry.symbol &&
-                    varSymbol.owner == resolvedSymbolEnv.scope.owner) {
+            if (entry != NOT_FOUND_ENTRY && varSymbol == entry.symbol) {
```

Comment:
Do we need this because we don't shadow the symbol like we discussed offline?

Reply:
Yes, when looking for the resolved level of a symbol, we don't need to check for the symbol owner. If there's the same symbol with two different owners, it becomes a compile-time error. I believe we introduced this logic before addressing the shadowing issue(that is the reason to add this check).

The reason for keeping the `varSymbol == entry.symbol` logic is because there are instances in the desugar where we use the same symbol generate variable. However, this is also not correct, and we need to address it separately.

Reply:
https://github.com/ballerina-platform/ballerina-lang/issues/42434